trussworks/react-uswds
# feat: Make search component extendable

# Summary

This PR:
- Adds `inputProps` to `Search` and passes it down to `TextInput`
- Split `Search` into two new components `SearchField` and `SearchButton`

## Related Issues or PRs

closes #2209  

## How To Test

1. Run `yarn storybook` and look at `Components/Search/*` notice the search input and search buttons have been broken out into their own components. And are composed together in `Search`
2. Run `yarn test:coverage` and verify that all tests are passing and that the coverage threshold is being met.


## Screenshots (optional)
### SearchField
<img width="929" alt="Screen Shot 2022-12-12 at 10 23 54 AM" src="https://user-images.githubusercontent.com/67110378/207084068-2d0210c4-4bf0-45d0-94f6-f12d54b4acf9.png">

### SearchButton
<img width="1011" alt="Screen Shot 2022-12-12 at 10 24 02 AM" src="https://user-images.githubusercontent.com/67110378/207084070-c19de7d6-4723-49fc-a6c3-185fd323da7d.png">

### Search
<img width="533" alt="Screen Shot 2022-12-12 at 10 24 08 AM" src="https://user-images.githubusercontent.com/67110378/207084071-fcfdb31c-d107-4c40-b75e-19c5e2fef226.png">
